3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/registry/MarketoAdapter.scala

```diff
@@ -0,0 +1,127 @@
+
+  private def payloadBodyToEvent(json: String, payload: CollectorPayload): Validated[RawEvent] = {
+
+    try {
+      val parsed = parse(json)
+
+      val parsed_converted = parsed transformField {
+        case ("acquisition_date", JString(value))         => ("acquisition_date", JString(JU.toJsonSchemaDateTime(value, MarketoDateTimeFormat)))
+        case ("created_at", JString(value))               => ("created_at", JString(JU.toJsonSchemaDateTime(value, MarketoDateTimeFormat)))
+        case ("email_suspended_at", JString(value))       => ("email_suspended_at", JString(JU.toJsonSchemaDateTime(value, MarketoDateTimeFormat)))
+        case ("last_referred_enrollment", JString(value)) => ("last_referred_enrollment", JString(JU.toJsonSchemaDateTime(value, MarketoDateTimeFormat)))
+        case ("last_referred_visit", JString(value))      => ("last_referred_visit", JString(JU.toJsonSchemaDateTime(value, MarketoDateTimeFormat)))
+        case ("updated_at", JString(value))               => ("updated_at", JString(JU.toJsonSchemaDateTime(value, MarketoDateTimeFormat)))
+        case ("datetime", JString(value))                 => ("datetime", JString(JU.toJsonSchemaDateTime(value, MarketoDateTimeFormat)))
+    }
+
+      val eventType = Some("event")
+
+      lookupSchema(eventType, VendorName, EventSchemaMap) map {
+        schema => RawEvent(
+            api = payload.api,
+            parameters = toUnstructEventParams(
+              TrackerVersion,
+              toMap(payload.querystring),
+              schema,
+              parsed_converted,
+              "srv"
+            ),
+            contentType = payload.contentType,
+            source = payload.source,
+            context = payload.context
+          )
+      }
+
+    } catch {
+      case e: JsonParseException => {
+        val exception = JU.stripInstanceEtc(e.toString).orNull
+        s"$VendorName event failed to parse into JSON: [$exception]".failNel
+      }
+    }
+  }
+
+  /**
+   * Converts a CollectorPayload instance into raw events.
+   * Marketo event contains no "type" field and since there's only 1 schema the function lookupschema takes the eventType parameter as "event".
+   * We expect the type parameter to match the supported events, else
+   * we have an unsupported event type.
+   *
+   * @param payload The CollectorPayload containing one or more
+   *        raw events as collected by a Snowplow collector
+   * @param resolver (implicit) The Iglu resolver used for
+   *        schema lookup and validation. Not used
+   * @return a Validation boxing either a NEL of RawEvents on
+   *         Success, or a NEL of Failure Strings
+   */
+  def toRawEvents(payload: CollectorPayload)(implicit resolver: Resolver): ValidatedRawEvents =
+    (payload.body, payload.contentType) match {
+      case (None, _) => s"Request body is empty: no ${VendorName} event to process".failNel
```

Comment:
Redundant curly braces around `VendorName`. `failNel` should be `failureNel` as former is deprecated.